Sprint 09

F1-KS opened the pull request.

F1-KS committed:
Начало sprint_09

F1-KS committed:
Добавил Tab Bar

F1-KS committed:
После обучения SingleViewController показывать разные картинки

F1-KS committed:
немного оптимизации кода

F1-KS committed:
перед внедрением ScrollView

F1-KS committed:
Добавил скролл

F1-KS committed:
Домашняя работа sprint_09

F1-KS committed:
немного поправил констрейнты

F1-KS committed:
корректировка кода в UIActivityViewController

protos17 commented on ImageFeed/ImagesList/ImagesListViewController.swift:
```diff
@@ -9,23 +9,36 @@ final class ImagesListViewController: UIViewController {
     
-}
+    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
+        if segue.identifier == ShowSingleImageSegueIdentifier { // Сначала мы проверяем идентификатор сегвея, поскольку может быть больше одного сегвея, выходящего из нашего контроллера
```
Рекомендую удалить все лишние комментарии

protos17 commented on ImageFeed/ImagesList/SingleImageViewController.swift:
```diff
@@ -0,0 +1,61 @@
+import Foundation
+import UIKit
+
```
Рекомендую удалить лишний отступ

protos17 commented on ImageFeed/ImagesList/SingleImageViewController.swift:
```diff
@@ -0,0 +1,61 @@
+import Foundation
```
Если импортирован UIKit, советую не испортировать Foundation, потому что UIKit содержит в себе Foundation

protos17 commented on ImageFeed/ImagesList/SingleImageViewController.swift:
```diff
@@ -0,0 +1,61 @@
+
+final class SingleImageViewController: UIViewController {
+    
+    @IBOutlet var imageView: UIImageView!
```
Необходимо указывать private для того, чтобы спрятать детали реализации конкретной части функциональности, когда они используются только внутри области объявления.

protos17 commented on ImageFeed/ImagesList/SingleImageViewController.swift:
```diff
@@ -0,0 +1,61 @@
+final class SingleImageViewController: UIViewController {
+    
+    @IBOutlet var imageView: UIImageView!
+    @IBAction func didTapShareButton(_ sender: Any) {
```
Необходимо указать private

protos17 commented on ImageFeed/ImagesList/SingleImageViewController.swift:
```diff
@@ -0,0 +1,61 @@
+        let sharingButton = UIActivityViewController(activityItems: [image as Any], applicationActivities: nil)
+        present(sharingButton, animated: true)
+    }
+    @IBOutlet var scrollView: UIScrollView!
```
Необходимо указать private

protos17 commented on ImageFeed/Profile/ProfileViewController.swift:
```diff
@@ -0,0 +1,7 @@
+
+
+final class ProfileViewController: UIViewController {
+    
```
Вот здесь необходимо создан кодом элементы и расставить им констрейнты

F1-KS committed:
Домашнее задание sprint_09, версия 2
IB_Profile оставил для себя
Описания тоже оставил для себя, понимаю что его надо писать короче.
